src/GafferArnold/ArnoldShader.cpp

```diff
@@ -128,12 +128,37 @@ void ArnoldShader::loadShader( const std::string &shaderName, bool keepExistingV
 
 	const bool isLightShader = AiNodeEntryGetType( shader ) == AI_NODE_LIGHT;
 	namePlug()->setValue( AiNodeEntryGetName( shader ) );
-	typePlug()->setValue(
-		isLightShader ? "ai:light" : "ai:surface"
-	);
+
+	int aiOutputType = AI_TYPE_POINTER;
+	string type = "ai:light";
+	if( !isLightShader )
+	{
+		bool isLightFilter = false;
+		const CompoundData *metadata = ArnoldShader::metadata();
+		if( metadata )
```

Comment:
Is it actually possible to get NULL metadata? As far as I can tell it isn't, in which case let's not defend against it.